[Bug]: Missing artifacts in staged data-tck-1.0.0.zip

### Specification Version

1.0.0-RC1

### Bug report

This is for the 1.0.0 stage TCK, 
https://www.eclipse.org/downloads/download.php?file=/ee4j/data/jakartaee/staged/eftl/data-tck-1.0.0.zip

I'm trying to update the Hibernate CCR, but I cannot run the artifact-install.sh due to missing artifacts:
```bash
starksm@Scotts-Mac-Studio artifacts % bash artifact-install.sh 1.0.0
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- install:3.0.0-M1:install-file (default-cli) @ standalone-pom ---
[ERROR] The specified file '/private/tmp/data-tck-1.0.0/artifacts/jakarta.data-parent-1.0.0.pom' not exists
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.197 s
[INFO] Finished at: 2024-04-30T09:18:46-06:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-install-plugin:3.0.0-M1:install-file (default-cli) on project standalone-pom: The specified file '/private/tmp/data-tck-1.0.0/artifacts/jakarta.data-parent-1.0.0.pom' not exists -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< org.apache.maven:standalone-pom >-------------------
[INFO] Building Maven Stub Project (No POM) 1
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- install:3.0.0-M1:install-file (default-cli) @ standalone-pom ---
[INFO] Installing /private/tmp/data-tck-1.0.0/artifacts/jakarta.data-tck-1.0.0.jar to /Users/starksm/.m2/repository/jakarta/data/jakarta.data-tck/1.0.0/jakarta.data-tck-1.0.0.jar
[INFO] Installing /var/folders/yh/t0fnl559607crz2vyp509bjr0000gn/T/jakarta.data-tck-1.0.018295126351470739934.pom to /Users/starksm/.m2/repository/jakarta/data/jakarta.data-tck/1.0.0/jakarta.data-tck-1.0.0.pom
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.218 s
[INFO] Finished at: 2024-04-30T09:18:47-06:00
[INFO] ------------------------------------------------------------------------
starksm@Scotts-Mac-Studio artifacts % ls
artifact-install.sh		jakarta.data.sig_17
jakarta.data-tck-1.0.0.jar	jakarta.data.sig_21
```

If I build the 1.0.0 tag of the data repo locally and add the jakarta.data-tck-1.0.0.pom to the artifacts directory the script runs correctly.



### Additional information

I see the tck-dist/src/main/assembly/assembly.xml includes the jakarta.data:jakarta.data-parent in its dependencySet, but this is not being included as if I look at the dist zip built from the 1.0.0 tag it does not include the jakarta.data-tck-1.0.0.pom:

```bash
starksm@Scotts-Mac-Studio tck-dist % jar -tf target/data-tck-1.0.0-dist.zip | grep artifacts
data-tck-1.0.0/artifacts/
data-tck-1.0.0/artifacts/artifact-install.sh
data-tck-1.0.0/artifacts/jakarta.data.sig_17
data-tck-1.0.0/artifacts/jakarta.data.sig_21
data-tck-1.0.0/artifacts/jakarta.data-tck-1.0.0.jar
```

I'll look at determining what needs to be updated to fix this.

--- a/tck-dist/pom.xml
+++ b/tck-dist/pom.xml
@@ -199,23 +199,6 @@
      </execution>
     </executions>
    </plugin>
-   <plugin>
-    <groupId>org.apache.maven.plugins</groupId>
-    <artifactId>maven-javadoc-plugin</artifactId>
-    <version>${maven-javadoc-plugin.vesion}</version>
-
-    <executions>
-     <execution>
-      <id>attach-javadocs</id>
-      <goals>
-       <goal>jar</goal>
-      </goals>
-      <configuration>
-       <skippedModules>jakarta.data-parent</skippedModules>
-      </configuration>
-     </execution>
-    </executions>
-   </plugin>
   </plugins>
  </build>
 </project>

--- a/tck-dist/pom.xml
+++ b/tck-dist/pom.xml
@@ -60,7 +60,7 @@
    <artifactId>jakarta.data-tck</artifactId>
    <version>${jakarta.data.tck.version}</version>
   </dependency>
-    <!-- Provided dependencies for the TCK -->
+  <!-- Provided dependencies for the TCK -->
   <dependency>
     <groupId>jakarta.servlet</groupId>
     <artifactId>jakarta.servlet-api</artifactId>

--- a/tck-dist/src/main/assembly/assembly.xml
+++ b/tck-dist/src/main/assembly/assembly.xml
@@ -36,6 +36,11 @@
       <source>src/main/readme/README.md</source>
       <destName>README.md</destName>
     </file>
+    <!-- The jakarta.data-parent pom file -->
+    <file>
+      <source>../pom.xml</source>
+      <destName>artifacts/jakarta.data-parent-${project.version}.pom</destName>
+    </file>
   </files>
 
   <fileSets>
@@ -77,7 +82,6 @@
       <includes>
         <include>jakarta.data:jakarta.data-tck</include>
         <include>jakarta.data:jakarta.data-tck:jar:sources</include>
-        <include>jakarta.data:jakarta.data-parent</include>
       </includes>
       <useTransitiveDependencies>true</useTransitiveDependencies>
       <outputDirectory>artifacts</outputDirectory>

